jax/_src/numpy/array_methods.py

```diff
@@ -800,13 +790,26 @@ def maybe_setattr(attr_name, target):
 def _set_device_array_attributes(device_array):
   setattr(device_array, "__array_module__", __array_module__)
 
+def _make_abstract_method(func):
+  return abc.abstractmethod(wraps(func)(lambda *args, **kwargs: func(*args, **kwargs)))
+
+def _set_array_abstract_methods(basearray):
+  for operator_name, function in _array_operators.items():
+    setattr(basearray, f"__{operator_name}__", _make_abstract_method(function))
+  for method_name, method in _array_methods.items():
+    setattr(basearray, method_name, _make_abstract_method(method))
+  for prop_name, prop in _array_properties.items():
+    setattr(basearray, prop_name, property(_make_abstract_method(prop)))
 
 def register_jax_array_methods():
   """Call this function once to register methods of JAX arrays"""
   _set_shaped_array_attributes(core.ShapedArray)
   _set_shaped_array_attributes(core.DShapedArray)
 
   _set_device_array_base_attributes(ArrayImpl, exclude={'__getitem__'})
+  _set_device_array_base_attributes(core.Tracer, exclude=_impl_only_array_methods)
```

Comment:
While you are here.. maybe rename this? (device_array doesn't exist anymore)

Reply:
Done